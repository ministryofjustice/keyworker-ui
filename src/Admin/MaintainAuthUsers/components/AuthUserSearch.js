import React from 'react'
import PropTypes from 'prop-types'
import InputField from '@govuk-react/input-field'
import Button from '@govuk-react/button'
import GridRow from '@govuk-react/grid-row'
import GridCol from '@govuk-react/grid-col'
import { lookupMeta } from '../../../govuk-helpers'
import { errorType } from '../../../types'

const AuthUserSearch = ({ user, handleChange, handleSearch, error }) => (
  <GridRow>
    <GridCol setWidth="two-thirds">
      <form onSubmit={handleSearch}>
        <InputField
          meta={lookupMeta('user', error)}
          htmlFor="user"
          mb={6}
          input={{ id: 'user', name: 'user', defaultValue: user, onChange: handleChange }}
        >
          Username or email address
        </InputField>
        <Button type="submit" id="search-button">
          Search
        </Button>
      </form>
    </GridCol>
  </GridRow>
)

AuthUserSearch.propTypes = {
  user: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleSearch: PropTypes.func.isRequired,
  error: errorType,
}

AuthUserSearch.defaultProps = {
  error: [],
}

export default AuthUserSearch
