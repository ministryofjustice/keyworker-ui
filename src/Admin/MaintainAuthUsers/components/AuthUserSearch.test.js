import React from 'react'
import renderer from 'react-test-renderer'
import AuthUserSearch from './AuthUserSearch'

describe('Auth search form', () => {
  const stubFunc = () => {}
  describe('Form', () => {
    it('should render correctly with no user', () => {
      const wrapper = renderer
        .create(<AuthUserSearch handleChange={stubFunc} handleSearch={stubFunc} user="" />)
        .toJSON()
      expect(wrapper).toMatchSnapshot()
    })
    it('should render correctly with a user', () => {
      const wrapper = renderer
        .create(<AuthUserSearch handleChange={stubFunc} handleSearch={stubFunc} user="someuser" />)
        .toJSON()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
