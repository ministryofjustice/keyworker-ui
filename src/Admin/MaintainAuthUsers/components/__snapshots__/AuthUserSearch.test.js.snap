// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth search form Form should render correctly with a user 1`] = `
<div
  className="sc-gzVnrw kFjkBp"
>
  <div
    className="sc-htoDjs cFfunZ"
  >
    <form
      onSubmit={[Function]}
    >
      <label
        className="sc-bdVaJa iHrfuG"
        htmlFor="user"
      >
        <span
          className="sc-bwzfXH eLZbP"
        >
          Username or email address
        </span>
        <input
          className="sc-ifAKCX hBHdQq"
          defaultValue="someuser"
          id="user"
          name="user"
          onChange={[Function]}
          type="text"
        />
      </label>
      <button
        className="sc-EHOje ihKsGU"
        disabled={false}
        id="search-button"
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</div>
`;

exports[`Auth search form Form should render correctly with no user 1`] = `
<div
  className="sc-gzVnrw kFjkBp"
>
  <div
    className="sc-htoDjs cFfunZ"
  >
    <form
      onSubmit={[Function]}
    >
      <label
        className="sc-bdVaJa iHrfuG"
        htmlFor="user"
      >
        <span
          className="sc-bwzfXH eLZbP"
        >
          Username or email address
        </span>
        <input
          className="sc-ifAKCX hBHdQq"
          defaultValue=""
          id="user"
          name="user"
          onChange={[Function]}
          type="text"
        />
      </label>
      <button
        className="sc-EHOje ihKsGU"
        disabled={false}
        id="search-button"
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</div>
`;
